### DISCUSSION TITLE
Issues with warehouses, granaries, taverns and fussy market ladies :D

### DISCUSSION THREAD

**Discussion by BazzaCuda** [Keriew/augustus#573] [Q&A] [answered]
I'm running v3.1.0

1. Despite the warehouse instruction, "Allow market ladies to buy from here", they never buy food, not even wheat. They will only buy wheat, fish, veg and fruit from a granary.

2. Problem: I am importing fruit by sea into a warehouse by the dock, set to accepting fruit. I have a granary near my luxury palaces set to "Getting fruit" but the granary's cart pusher stays put; the cart pusher from the dock's warehouse, however, goes out on delivery taking one [100] cartload at a time from the warehouse to the granary. If I swap the granary for a warehouse, the "getting" all works as expected (the cart pusher getting 400) except that the market ladies will ignore all the fruit sat in the warehouse right next to the market!

3. Taverns will buy wine from a warehouse but not fish. Taverns will only buy fish from a granary. Same as market ladies.

4. I have a warehouse full of Veg, Fruit, Wheat and Fish, directly opposite a market, and the market buyer is still "looking for a source of food to sell". If I transfer the entire stock to a granary, the market buyer lady will buy (and distribute) the lot! I think someone needs to have a word with her!! :D

5. Finally, the warehouse/granary setting "Allow market ladies to buy from here" is a bit of a misnomer in that this setting also applies to Taverns and Caravanserais (and others?), so might be more accurate as "local traders" in contrast to the non-local traders (caravans, ships) ?

Conclusions:
My understanding is that "getting" should always be carried out by the cart pusher from the getting location.
Market ladies should be able to buy, from any source, all the commidities that markets sell, whether that's from a granary or a warehouse, if it's in stock, no? (same for Taverns)

Baz.

**Comment by crudelios** [comment]
This is by design, since the original C3. As a rule of thumb, you shouldn't place food on warehouses. Food on warehouses only serve two purposes:

1. Exporting/ importing
2. Caesar requests

With augustus, you can even import/ export/ fulfill requests directly from granaries (you'll need to set the proper options though), so there is no reason at all to use warehouses for storing food.

I don't fully remember from the code, but I think market ladies will only get food from warehouses if there is no other option available at all or if the food is imported. Though I'm not sure of even that.

**Comment by BazzaCuda** [comment]
Thanks for the prompt reply, I'll stop expecting my markets to buy food from warehouses.
I can confirm that the market ladies won't buy food from a warehouse even when it's the only available food source.

That still leaves point 2 though, where my granary isn't "getting" the fruit from the dock's warehouse 400 units at a time - rather, the dock is delivering 100 units to the granary. This doesn't seem correct (?).

**Comment by crudelios** [comment, marked as answer]
I don't think granaries get from warehouses, they only get from other granaries. I guess we could change that. Frankly as the code is right now, there's little point in allowing warehouses to store food actually...

**Reply by BazzaCuda** [reply]
I think, then, that I've identified my misunderstanding and the reason why my food supply chain was failing: I've always been in the habit of importing/exporting from warehouses next to docks. Now that I need a third, imported, food source, aka fruit, I should have put a granary next to the dock, which other granaries can then "get" from and which my market ladies will buy from. Thanks for your help; I got to the nub of my problem in the end :D

**Reply by crudelios** [reply]
No problem, this is a gameplay change we added in augustus to make handling food less cumbersome. I guess we should explain the changes better!

**Reply by BazzaCuda** [reply]
haha, definitely! :D
One thing that is very misleading, in my opinion, is the warehouse instruction, "Allow market ladies to buy from here". I think it should be, "Allow locals to buy from here". If you have one warehouse full of timber with this option unchecked, a lighthouse never gets any timber and yet its not the market ladies that collect/deliver the timber to the lighthouse. I can't work out if the lighthouse (like taverns) sends out someone to the warehouse or if it's the warehouse cartpusher who delivers to the [demanding] lighthouse, but I'm fairly certain it's not the market ladies :D....and timber yards don't deliver direct to a lighthouse. It took me a while to figure that one out.
